### DISCUSSION TITLE
Compile fails on imgui_impl_dx12.cpp

### DISCUSSION THREAD

**Discussion by vasitzin** [ocornut/imgui#5596] [New Users Build/Link/Run issues ONLY!]
Hi everybody, so I'm encountering an issue with my imgui project.

*I wasn't quite sure if I should create an issue or a discussion, but since I've never actually compiled any c++ project longer than 10 lines on Windows it's highly probable that there is something wrong with my setup.*

The compiler gives me an error on this [line](https://github.com/ocornut/imgui/blob/8cbd391f096b9314a08670052cc0025cbcadb249/backends/imgui_impl_dx12.cpp#L596) reading:
`'PFN_D3D12_SERIALIZE_ROOT_SIGNATURE' was not declared in this scope; did you mean 'PFN_D3D12_SERIALIZE_VERSIONED_ROOT_SIGNATURE'?`
I've tried following it's suggestion to no avail.

I'm compiling my win32+dx12 project on a Windows 10 VM with CMake, mingw64-make, mingw64-gcc. Note that with dx11 everything works just fine. Here is a gist summary of my [CMakeLists](https://gist.github.com/vasitzin/818a4734df131b4910fb79a985cf5274).
I'm new to Windows development so any help or tip is appreciated, thanks in advance.

**Comment by ocornut** [comment]
This appears to be the same as  #4594 #4604 which is still unsolved.

**Reply by vasitzin** [reply]
Thank you for pointing these out I should have found them myself.
Having read through them I agree with the sentiment that there's nothing to be fixed on imgui's side so I'll just implement your #ifdef mingw solution on my fork just to get it working for now.
